Samples/Notifications/App/CppUnpackagedAppNotifications/CppUnpackagedAppNotifications/Notifications/ToastWithAvatar.cpp

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ToastWithAvatar.h"
+#include "Common.h"
+#include <winrt/Microsoft.Windows.AppNotifications.h>
+#include "App.xaml.h"
+#include "MainPage.xaml.h"
+
+namespace winrt
+{
+    using namespace Microsoft::Windows::AppNotifications;
+    using namespace CppUnpackagedAppNotifications::implementation;
+}
+
+const wchar_t* ToastWithAvatar::ScenarioName{ L"Local Toast with Avatar Image" };
+
+bool ToastWithAvatar::SendToast()
+{
+    // The ScenarioIdToken uniquely identify a scenario and is used to route the response received when the user clicks on a toast to the correct scenario.
+    auto ScenarioIdToken{ Common::MakeScenarioIdToken(ToastWithAvatar::ScenarioId) };
+
+    winrt::hstring xmlPayload{
+        L"<toast launch = \"action=ToastClick&amp;" + ScenarioIdToken + L"\">\
+            <visual>\
+                <binding template = \"ToastGeneric\">\
+                    <image placement = \"appLogoOverride\" hint-crop=\"circle\" src = \"" + winrt::App::GetFullPathToAsset(L"Square150x150Logo.png") + L"\"/>\
+                    <text>" + ScenarioName + L"</text>\
```

Comment:
You can avoid a lot of the L" and double "" by doing this:

winrt::hstring payload =
        LR"(<toast launch="action = viewDownload &amp; downloadId = 9438108">
        <visual>
            <binding template = "ToastGeneric">
                <text>Downloading this week's new music...</text>
                <progress
                    title = "{progressTitle}"
                    value = "{progressValue}"
                    valueStringOverride = "{progressValueString}"
hulumane marked this conversation as resolved.
hulumane marked this conversation as resolved.
                    status = "{progressStatus}" />
            </binding>
        </visual>
    </toast>)";

Reply:
https://github.com/microsoft/WindowsAppSDK/pull/1675/files#:~:text=winrt%3A%3Ahstring%20payload,%3C/toast%3E)%22%3B